### ISSUE TITLE
CA1081 (unused parameter) should be suppressed for main method

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn-analyzers#2174]
#### Analyzer package

[Microsoft.CodeQuality.Analyzers](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/)

#### Package Version

v2.6.3

#### Diagnostic ID

[CA1801](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1801-review-unused-parameters?view=vs-2017#feedback)

#### Repro steps

```csharp
public static class Program
{
	public static void Main(string[] args)
        {
                 Console.WriteLine("Hello World");
        }
}
```

#### Expected behavior

No warning. This the common pattern Main methods follow, and what visual studio generates by default.

#### Actual behavior

Warning CA1081 Parameter args of method Main is never used. Remove the parameter or use it in the method body.

**Comment by mavasani** [comment]
This is by design - the idea is to flag all valid cases and let the user decide to suppress if they don't care about the violation or see a false positive. This also matches the behavior of legacy post-build FxCop.

> visual studio generates by default

The default template has unused usings, parameter, etc. so that the user is aware about code that they can write, such as ability to have command line arguments. Unused parameters warning and greying out the unused usings allows the developer to make a choice whether to delete the code or use it in some way.
